src/lib/common/topology_sampler.cpp

```diff
@@ -611,6 +607,23 @@ static bool encode_cbor_configuration(CborEncoder &array_encoder,
           }
           break;
         }
+        case CborArrayType: {
+          // To determine how many bytes long the cbor array is,
+          // we need to advance the iterator past the end of the array.
+          // Once we know the encoded byte length, we can simply memcpy.
+          // We then need to skip the cbor_value_advance call at the end of the loop,
+          // since we will have already advanced the iterator.
+          const uint8_t *array_start = it.source.ptr;
+          already_advanced = true;
+          err = cbor_value_advance(&it);
+          if (err != CborNoError) {
+            break;
+          }
+          size_t encoded_array_byte_len = it.source.ptr - array_start;
+          memcpy(map_encoder.data.ptr, array_start, encoded_array_byte_len);
+          map_encoder.data.ptr += encoded_array_byte_len;
+          break;
+        }
```

Comment:
Per your feedback @russelldeguzman here's the array copy in the topology sampler.